lib/ecl/tests/ecl_nnc_export_intersect.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+   Copyright (C) 2018  Statoil ASA, Norway.
+
+   The file 'ecl_nnc_export_intersect.c' is part of ERT - Ensemble based
+   Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more detals.
+*/
+// #include <stdlib.h>
+// #include <stdbool.h>
+#include <signal.h>
+
+#include <ert/util/test_util.hpp>
+
+#include <ert/ecl/ecl_nnc_export.hpp>
+#include <ert/ecl/ecl_kw_magic.hpp>
+#include <ert/ecl/ecl_nnc_data.hpp>
+#include <ert/ecl/ecl_grid.hpp>
+#include <ert/ecl/ecl_file.hpp>
+
+#include <ert/util/ert_unique_ptr.hpp>
+#include <ert/util/util.hpp>
+
+
+
+namespace {
+
+const auto GRIDX = 10, GRIDY = 10, GRIDZ = 10;
+const auto GRID_NNC_NUM = 342;
+const auto INIT_NNC_NUM = 298;
+
+
+void install_SIGNALS(void) {
+  signal(SIGSEGV , util_abort_signal);    /* Segmentation violation, i.e. overwriting memory ... */
+  signal(SIGINT  , util_abort_signal);    /* Control C */
+  signal(SIGTERM , util_abort_signal);    /* If killing the program with SIGTERM (the default kill signal) you will get a backtrace.
+                                             Killing with SIGKILL (-9) will not give a backtrace.*/
+}
+
+
+ERT::ert_unique_ptr<ecl_grid_type, ecl_grid_free>
+make_intersect_grid() {
+   auto out = ERT::ert_unique_ptr<ecl_grid_type, ecl_grid_free>(
+         ecl_grid_alloc_rectangular(GRIDX, GRIDY, GRIDZ, 1., 1., 1., nullptr)
+         );
+   for(auto i = 0; i < GRID_NNC_NUM; ++i)
```

Comment:
This is at least invalid from a domain point of view; although the test might actually work. I would suggest something like:
```
for (auto i=0; i < GRID_NNC_NUM; i++)
     ecl_grid_add_self_nnc(out.get(), 2*i, 2*i + 1, i);
```